src/db.rs

```diff
@@ -120,11 +120,13 @@ struct DbInner {
 }
 
 impl DbInner {
-	fn open(options: &Options) -> Result<DbInner> {
-		std::fs::create_dir_all(&options.path)?;
+	fn open(options: &Options, create: bool) -> Result<DbInner> {
+		if create {
+			std::fs::create_dir_all(&options.path)?
+		};
 		let mut lock_path: std::path::PathBuf = options.path.clone();
 		lock_path.push("lock");
-		let lock_file = std::fs::OpenOptions::new().create(true).read(true).write(true).open(lock_path.as_path())?;
+		let lock_file = std::fs::OpenOptions::new().create(create).read(true).write(true).open(lock_path.as_path())?;
```

Comment:
Creation of lock file should be attempted regardless of the `create` flag. I.e. when opening an existing database with no lock, it should be created and taken.

Reply:
:+1: (the locks usually exists but being able to remove such file is something expected)